### PR TITLE
Remove git2 as a dependency.

### PR DESCRIPTION
**Pull request by calavera**
This dependency cannot be sent between threads.
Use git-config which is a pure rust implementation to read git config files.

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: lib/src/ignore/files.rs [hunk 1]
```diff
@@ -62,18 +64,26 @@ pub async fn from_origin(path: impl AsRef<Path>) -> (Vec<IgnoreFile>, Vec<Error>
...
-					discover_file(
-						&mut files,
-						&mut errors,
-						None,
-						Some(ProjectType::Git),
-						excludes,
-					)
-					.await;
+				if let Ok(excludes) = config.value::<Cow<[u8]>>("core", None, "excludesFile") {
+					match String::from_utf8(excludes.into_owned()) {
```
Comment (passcod) [inline lib/src/ignore/files.rs:71]: I'm pretty sure you can go from a Cow<u8> to a Path without failing on non-utf8 paths, which is something I'd rather avoid.

File: lib/src/ignore/files.rs [hunk 2]
```diff
@@ -1,4 +1,6 @@
+use git_config::file::GitConfig;
```
Comment (passcod) [inline lib/src/ignore/files.rs:1]: This should be `ResolvedGitConfig` as per the [docs](https://docs.rs/git-config/0.1.11/git_config/file/struct.GitConfig.html)

**Review by passcod — CHANGES_REQUESTED** [review]
Looks great! Also that should avoid issues with incompatible libgit2 library versions, which happen every so often... Only one change I'd like made

File: lib/src/ignore/files.rs (hunk 1 above)
Comment (calavera) [reply lib/src/ignore/files.rs:71]: It's actually not as easy as it looks. The `git-config` crate has a Path type that can do that "automatically", but it has not been released in a stable version yet. I'll ping the maintainers to see when they are planning a release, and will update this accordingly.

**Review by passcod — APPROVED** [review]

**Comment by passcod** [comment]
Rebase and I'll merge it :)

**Comment by calavera** [comment]
@passcod rebased

**Comment by passcod** [comment]
Released as 2.0.0-pre.14

**Pull request merged**